### PR TITLE
wip CRM-19815 contribution search performance fixes

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
* [CRM-19815: Remove performance degrading-joins from civicrm_contribution search](https://issues.civicrm.org/jira/browse/CRM-19815)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@monishdeb @seamuslee001 I think you too are the only ones familiar enough with the search code (& the risks on changing it) to review this - I can do some quid-pro-quo review on this one as it will save wmf code review time.

**Comment by jitendrapurohit** [comment]
jenkins, test this please

**Comment by eileenmcnaughton** [comment]
Ok - that's a legit breakage - failure to account for a legacy param. Will take a look

**Comment by eileenmcnaughton** [comment]
this works except order by, but I want to work the metadata better so doing #9631 first

File: CRM/Contribute/BAO/Query.php [hunk 1]
```diff
@@ -975,7 +941,7 @@ public static function defaultReturnProperties($mode, $includeCustomFields = TRU
         // actually
         'is_pay_later' => 1,
         // required
-        'contribution_status' => 1,
+        // 'contribution_status' => 1,
```
Comment (monishdeb) [inline CRM/Contribute/BAO/Query.php]: umm why we are ignoring pseudo attributes ?

Comment (monishdeb) [inline CRM/Contribute/BAO/Query.php]: ahh got it .. so we ONLY show pseudo attribute if asked in ```return``` not by default.

Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Query.php]: well once the pseudo attributes work correctly they automatically add contribution_status when contribution_status_id is requested  - so it's redundant

**Comment by eileenmcnaughton** [comment]
Closing in favour of #9631

**Pull request closed without merging**